app/battle.py

```diff
@@ -0,0 +1,39 @@
+from app.knight import Knight
+
+
+class Battle:
+    battle_result = {}
+
+    def __init__(
+            self,
+            first_knight: Knight,
+            second_knight: Knight
+    ) -> None:
+        self.first_knight = first_knight
+        self.second_knight = second_knight
+
+    def battle_round(self) -> None:
+        first_knight_hp = (self.first_knight.hp - (
+            self.second_knight.power - self.first_knight.protection
+        ))
+        second_knight_hp = (self.second_knight.hp - (
+            self.first_knight.power - self.second_knight.protection
+        ))
```

Comment:
The calculation for `first_knight_hp` and `second_knight_hp` might result in negative values if the opponent's power is greater than the knight's protection. This is handled later by setting the HP to 0 if it's less than or equal to 0, which is correct according to the task description. Ensure this logic aligns with your intended game mechanics.